Services/TMS/ReportUtilities/classes/class.ilUDFWrapper.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+use ILIAS\TMS\Filter;
+use ILIAS\TMS\TableRelations;
+
+/**
+ * Provide access to ILIAS' UDFs.
+ *
+ * getLUAVisibleFields will return a dict (field_id=>field_name) of UDFs.
+ * To use this, setup your space and append UDFs with appendUDFs();
+ * this will extend $space with a DerivedTable (id='udf').
+ * Example:
+ *		$space = $this->appendUDFs(
+ *			$this->tf, $this->pf,
+ *			$space, $usr_data, 'usr_id'
+ *		);
+ *
+ * For output, you will have to adjust the row-template;
+ * add a generic block like this:
+ *		<!-- BEGIN udf_block -->
+ *			<td>{VAL_UDFFIELD}</td>
+ *		<!-- END udf_block -->
+ *
+ * Finally, you will have to define colums (defineFieldColumn):
+ *  	$column_id = 'UDF_' .(string)$field_id;
+ *		$table = $table->defineFieldColumn(
+ *			$fieldname,
+ *			$column_id,
+ *			[$column_id => $space->table('udf')->field($fieldname)]
+ *			,true
+ *		);
+ *
+ * Note, that the column-id starts with 'UDF_'.
+ * The fillRow-method will check for this and touch the udf_block.
+ * This way, there is no need to revise the template when adding fields.
+ *
+ * @author Nils Haagen 	<nils.haagen@concepts-and-traning.de>
+ */
+trait ilUDFWrapper {
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getLUAVisibleFields() {
```

Comment:
Ich brauche das selten, oder? Dann bitte in `getFieldsVisibleInLocalUserAdministration`, oder so ähnlich, umbenennen, jedenfalls keine Abkürzung. Die verstehe ich nämlich nur, wenn ich in die Methode reingucke.

Reply:
Brauche ich die Methode außerhalb des Traits?